numpy/core/src/common/npy_ctypes.h

```diff
@@ -0,0 +1,46 @@
+#ifndef NPY_CTYPES_H
+#define NPY_CTYPES_H
+
+#include <Python.h>
+
+#include "npy_import.h"
+
+/*!
+ * Check if a python type is a ctypes class
+ *
+ * This entire function is just a wrapper around the Python function of the
+ * same name.
+ */
+NPY_INLINE static int
+npy_ctypes_check(PyTypeObject *obj)
+{
+    static PyObject *py_func = NULL;
+    PyObject *ret_obj;
+    int ret;
+
+    npy_cache_import("numpy.core._internal", "npy_ctypes_check", &py_func);
+    if (py_func == NULL) {
+        goto fail;
+    }
+
+    ret_obj = PyObject_CallFunctionObjArgs(py_func, (PyObject *)obj, NULL);
+    if (ret_obj == NULL) {
+        goto fail;
+    }
+
+    ret = PyObject_IsTrue(ret_obj);
+    if (ret == -1) {
+        goto fail;
+    }
+
+    return ret;
+
+fail:
+    /* If the above fails, then we should just assume that the type is not from
+     * ctypes
+     */
+    PyErr_Clear();
+    return 0;
+}
+
+#endif
```

Comment:
new file needs to be added to `core/setup.py` `common_deps'

Reply:
Why doesn't CI pick up on that?

Reply:
I used `npy_import.h` as a template, and couldn't find it anywhere, so assumed I was in the clear.

Reply:
Hmm, that seems like it is missing as well. Updating missing files will not trigger compilation on `setup.py build_ext`. Try 

```
python setup.py build_ext
touch numpy/core/src/common/npy_import.h
python setup.py build_ext
```
Nothing gets built. Now try `touch numpy/core/src/common/mem_overlap.h`, everything gets rebuilt

Reply:
We could do a `check_all_files()` in `setup_common` that would run at build time, like `check_api_version` and refuse to build if unkown files are in `src/*`

Reply:
Updated to add this header and npy_import to common_deps - thanks for noticing that!